src/utils/resolve.ts

```diff
@@ -103,6 +104,8 @@ function isValidNewResolver(resolver: unknown): resolver is NewResolver {
   return true
 }
 
+let nodeResolverInstanceForLegacyNodeResolverSettings: NewResolver | null = null
```

Comment:
Why not defaults to `undefined`?

Reply:
Done in [`7dab322` (#272)](https://github.com/un-ts/eslint-plugin-import-x/pull/272/commits/7dab322c2fc41bbf101f249acf0549aab1c15e2b)